catroid/src/main/java/org/catrobat/catroid/formulaeditor/FormulaElement.java

```diff
@@ -315,62 +317,42 @@ private Object interpretUserList(Sprite sprite) {
 		if (userListValues.size() == 0) {
 			return EMPTY_USER_LIST_INTERPRETATION_VALUE;
 		} else if (userListValues.size() == 1) {
-			Object userListValue = userListValues.get(0);
-			if (userListValue instanceof String) {
-				return userListValue;
-			} else {
-				return userListValue;
-			}
+			return userListValues.get(0);
 		} else {
 			return interpretMultipleItemsUserList(userListValues);
 		}
 	}
 
 	private Object interpretMultipleItemsUserList(List<Object> userListValues) {
 		List<String> userListStringValues = new ArrayList<>();
-		boolean concatenateWithoutWhitespace = true;
+		boolean whiteSpaceNeeded = false;
 
 		for (Object listValue : userListValues) {
 			if (listValue instanceof Double) {
 				Double doubleValueOfListItem = (Double) listValue;
-				if (isNumberAIntegerBetweenZeroAndNine(doubleValueOfListItem)) {
-					userListStringValues.add(doubleValueOfListItem.intValue() + "");
-				} else {
-					concatenateWithoutWhitespace = false;
-					userListStringValues.add(listValue.toString());
-				}
+				userListStringValues.add(stringWithoutTrailingZero(String.valueOf(doubleValueOfListItem.intValue())));
 			} else if (listValue instanceof String) {
 				String stringValueOfListItem = (String) listValue;
-				if (stringValueOfListItem.length() == 1) {
-					userListStringValues.add(stringValueOfListItem);
-				} else {
-					userListStringValues.add(stringValueOfListItem);
-					concatenateWithoutWhitespace = false;
-				}
+				userListStringValues.add(stringWithoutTrailingZero(stringValueOfListItem));
 			}
 		}
 
 		StringBuilder stringBuilder = new StringBuilder(userListStringValues.size());
-		boolean isFirstListItem = true;
 		for (String userListStringValue : userListStringValues) {
```

Comment:
to make this a little bit easier to read - 
if youd move this for loop in a method that is called "listConsistsOfSingleCharacters" or similar, and then instead of using the boolean flag "whiteSpaceNeeded" add a new variable named "seperator" and before concatinting the list you set the separator to either "" or " " based on the return of above method

Reply:
done.